ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/MultiLineLoopRuleTest.kt

```diff
@@ -0,0 +1,229 @@
+            fun foo() {
+                for (i in 1..10) {
+                    return 0
+                }
+                while (true) {
+                    return 0
+                }
+                do {
+                    return 0
+                } while (true)
+            }
+            """.trimIndent()
+        multiLineLoopRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(3, 9, "Missing { ... }"),
+                LintViolation(5, 9, "Missing { ... }"),
+                LintViolation(7, 9, "Missing { ... }"),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given deep nested loop statements without curly braces`() {
+        val code =
+            """
+            fun main() {
+                for (i in 1..10)
+                    while (true)
+                        do
+                            return 0
+                        while (true)
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun main() {
+                for (i in 1..10) {
+                    while (true) {
+                        do {
+                            return 0
+                        } while (true)
+                    }
+                }
+            }
+            """.trimIndent()
+        multiLineLoopRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(3, 9, "Missing { ... }"),
+                LintViolation(4, 13, "Missing { ... }"),
+                LintViolation(5, 17, "Missing { ... }"),
+            ).isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given loop statements inside a lambda`() {
+        val code =
+            """
+            fun test(s: String?): Int {
+                val i = s.let {
+                    for (i in 1..10)
+                        1
+                    while (true)
+                        2
+                    do
+                        3
+                    while (true)
+                } ?: 0
+                return i
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun test(s: String?): Int {
+                val i = s.let {
+                    for (i in 1..10) {
+                        1
+                    }
+                    while (true) {
+                        2
+                    }
+                    do {
+                        3
+                    } while (true)
+                } ?: 0
+                return i
+            }
+            """.trimIndent()
+        multiLineLoopRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(4, 13, "Missing { ... }"),
+                LintViolation(6, 13, "Missing { ... }"),
+                LintViolation(8, 13, "Missing { ... }"),
+            ).isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given a do-while-statement with do keyword on same line as main statement`() {
```

Comment:
Please replace with `Given a do-while-statement with do keyword on same line as body statement and while keyword on separate line`